include/stdexec/execution.hpp

```diff
@@ -15,33 +15,40 @@
  */
 #pragma once
 
-#include <atomic>
-#include <cassert>
-#include <concepts>
-#include <condition_variable>
-#include <stdexcept>
-#include <memory>
-#include <mutex>
-#include <optional>
-#include <system_error>
-#include <tuple>
-#include <type_traits>
-#include <variant>
-
 #include "__detail/__execution_fwd.hpp"
 
+#include "__detail/__config.hpp"
+#include "__detail/__type_traits.hpp"
 #include "__detail/__env.hpp"
 #include "__detail/__domain.hpp"
 #include "__detail/__intrusive_ptr.hpp"
 #include "__detail/__meta.hpp"
 #include "__detail/__scope.hpp"
 #include "__detail/__basic_sender.hpp"
 #include "__detail/__utility.hpp"
+
 #include "functional.hpp"
 #include "concepts.hpp"
 #include "coroutine.hpp"
 #include "stop_token.hpp"
 
+#include <atomic>
+#include <cassert>
+#include <concepts>
+#include <condition_variable>
+#include <stdexcept>
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <system_error>
+#include <tuple>
+#include <type_traits>
+#include <variant>
+#include <coroutine>
```

Comment:
Note: This was suggested, but perhaps it should be omitted here, or guarded with the proper #ifdef